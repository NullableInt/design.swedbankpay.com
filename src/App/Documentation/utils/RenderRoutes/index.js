import React from "react";
import PropTypes from "prop-types";
import { Switch, Route, Redirect } from "react-router-dom";

const RenderRoutes = ({ path, redirect, routes }) => (
    <Switch>
        <Route exact path={path} render={() => <Redirect to={redirect} />}/>
        {routes.map((route, i) => {
            const { path, component } = route;

            return <Route key={i} exact path={path} render={() => <component.default />} />;
        })}
        <Redirect from={`${path}/*`} to="/404" />
    </Switch>
);

RenderRoutes.propTypes = {
    path: PropTypes.string.isRequired,
    redirect: PropTypes.string.isRequired,
    routes: PropTypes.array.isRequired
};

export default RenderRoutes;
