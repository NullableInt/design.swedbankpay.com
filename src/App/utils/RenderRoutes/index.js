import React from "react";
import PropTypes from "prop-types";
import { Switch, Route, Redirect } from "react-router-dom";
import Loadable from "react-loadable";

import { LoadingComponent } from "@src/App/utils";

const RenderRoutes = ({ path, redirect, routes, appFolder }) => (
    <Switch>
        <Route exact path={path} render={() => <Redirect to={redirect} />}/>
        {routes.map(route => {
            const { path, componentPath } = route;
            const LoadableComponent = Loadable({
                loader: () => import(/* webpackChunkName: "doc-route.chunk_" */ `../../${appFolder}/${componentPath}/index.js`),
                loading: LoadingComponent
            });

            return <Route key={`doc_route_${path}`} exact path={path} render={() => <LoadableComponent />} />;
        })}
        <Redirect from={`${path}/*`} to="/404" />
    </Switch>
);

RenderRoutes.propTypes = {
    path: PropTypes.string.isRequired,
    redirect: PropTypes.string.isRequired,
    routes: PropTypes.array.isRequired,
    appFolder: PropTypes.string.isRequired
};

export default RenderRoutes;
