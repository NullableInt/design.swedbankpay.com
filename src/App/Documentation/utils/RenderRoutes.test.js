import React from "react";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import RenderRoutes from "./RenderRoutes";

// TODO: Refactor and clean this up, write proper tests!! [EH]
describe("Utilities: RenderRoutes", () => {
    it("is defined", () => {
        expect(RenderRoutes).toBeDefined();
    });

    it("renders a basic route structure", () => {
        const props = {
            path: "test",
            redirect:"test",
            routes: [
                {
                    path: "test-2",
                    component: {
                        default: "test-2"
                    }
                }
            ]
        };

        const wrapper = mount(
            <MemoryRouter>
                <RenderRoutes {...props} />
            </MemoryRouter>
        );

        expect(wrapper).toMatchSnapshot();
    });
});

