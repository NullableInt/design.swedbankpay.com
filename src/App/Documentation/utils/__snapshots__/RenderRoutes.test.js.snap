// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utilities: RenderRoutes renders a basic route structure 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "z8r0xy",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "z8r0xy",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RenderRoutes
      path="test"
      redirect="test"
      routes={
        Array [
          Object {
            "component": Object {
              "default": "test-2",
            },
            "path": "test-2",
          },
        ]
      }
    >
      <Switch />
    </RenderRoutes>
  </Router>
</MemoryRouter>
`;
